File: linjiashop-admin/src/main/java/com/ityu/linjiaadminapi/controller/system/NoticeController.java
package com.ityu.linjiaadminapi.controller.system;


import com.google.common.base.Strings;
import com.ityu.bean.entity.system.Notice;
import com.ityu.service.system.NoticeService;
import com.ityu.bean.vo.front.Rets;
import com.ityu.web.controller.BaseController;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * NoticeController
 *
 * @author enilu
 * @version 2018/12/17 0017
 */
@RestController
@RequestMapping("/api/notice")
public class NoticeController extends BaseController {
    @Autowired
    private NoticeService noticeService;
    /**
     * 获取通知列表
     */
    @RequestMapping(value = "/list")
    public Object list(String condition) {
        List<Notice> list = null;
        if(Strings.isNullOrEmpty(condition)) {
            list =  noticeService.queryAll();
        }else{
            list = noticeService.findByTitleLike(condition);
        }
        return Rets.success(list);
    }
}

File: sdf-api/src/main/java/com/ityu/sdf/api/controller/system/NoticeController.kt
package com.ityu.sdf.api.controller.system

import com.ityu.common.bean.entity.system.Notice
import com.ityu.common.bean.vo.front.Ret
import com.ityu.common.bean.vo.front.Rets
import com.ityu.common.service.system.NoticeService
import io.swagger.annotations.Api
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * UserController
 *
 * @author enilu
 * @version 2018/9/15 0015
 */
@Api(tags = ["notice"])
@Tag(name = "notice", description = "通知模块")
@RestController
@RequestMapping("/notice")
open class NoticeController(
    private var noticeService: NoticeService,
) {

    /**
     * 获取通知列表
     */
    @GetMapping(value = ["/list"])
    open fun list(condition: String?): Ret<*> {
        var list: List<Notice?>? = null
        list = if (condition.isNullOrEmpty()) {
            noticeService.queryAll()
        } else {
            noticeService.findByTitleLike(condition)
        }
        return Rets.success(list)
    }


}
